src/app/donation.service.ts

```diff
@@ -71,24 +71,31 @@ export class DonationService {
     return couplet.donation;
   }
 
-  isResumable(donation: Donation): boolean {
-    return (donation.status !== undefined && this.resumableStatuses.includes(donation.status));
+  isResumable(donation: Donation, requiredPaymentMethodType: 'card' | 'customer_balance'): boolean {
+    return (
+      donation.status !== undefined &&
+      this.resumableStatuses.includes(donation.status) &&
+      donation.paymentMethodType === requiredPaymentMethodType
+    );
   }
 
   /**
    * Get a recent eligible-for-resuming donation to this same campaign/project, if any exists. We look
    * for candidates based on local status initially but return an Observable resulting from a re-GET
-   * so we include the latest server status. This is why this private method only knows the donation is
+   * so we include the latest server status. This is why this method only knows the donation is
    * 'probably' resumable.
    */
-  getProbablyResumableDonation(projectId: string): Observable<Donation | undefined> {
+  getProbablyResumableDonation(
+    projectId: string,
+    requiredPaymentMethodType: 'card' | 'customer_balance',
```

Comment:
Definitely not a blocker, but I'm not sure what the word `required` is doing for us here. IMHO `paymentMethodType` would be enough here and in `isResumable`.

Reply:
Yep, I suppose by the same logic as `projectId` it's superfluous. Updated.